docs/tutorials/audio.ipynb

```diff
@@ -0,0 +1,475 @@
+      "metadata": {
+        "colab_type": "text",
+        "id": "MUXex9ctTuDB"
+      },
+      "source": [
+        "## Setup"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "upgCc3gXybsA",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Install required Packages, and restart runtime"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "NwL3fEMQuZrk",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "try:\n",
+        "  # Use the Colab's preinstalled TensorFlow 2.x\n",
+        "  %tensorflow_version 2.x \n",
+        "except:\n",
+        "  pass"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "uUDYyMZRfkX4",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "!pip install tensorflow-io-nightly"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "J0ZKhA6s0Pjp",
+        "colab_type": "text"
+      },
+      "source": [
+        "## Usage"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "yZmI7l_GykcW",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Read an Audio File\n",
+        "\n",
+        "In TensorFlow IO, class `tfio.audio.AudioIOTensor` allows you to read an audio file into a lazy-loaded `IOTensor`:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "nS3eTBvjt-O5",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "import tensorflow as tf\n",
+        "import tensorflow_io as tfio\n",
+        "\n",
+        "audio = tfio.audio.AudioIOTensor('gs://cloud-samples-tests/speech/brooklyn.flac')\n",
+        "\n",
+        "print(audio)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "z9GCyPWNuOm7",
+        "colab_type": "text"
+      },
+      "source": [
+        "In the above example, the Flac file `brooklyn.flac` is from a publicly accessible audio clip in [google cloud](https://cloud.google.com/speech-to-text/docs/quickstart-gcloud).\n",
+        "\n",
+        "The GCS address `gs://cloud-samples-tests/speech/brooklyn.flac` are used directly because GCS is a supported file system in TensorFlow. In addition to `Flac` format, `WAV`, `Ogg`, `MP3`, and `MP4A` are also supported by `AudioIOTensor` with automatic file format detection.\n",
+        "\n",
+        "`AudioIOTensor` is lazy-loaded so only shape, dtype, and sample rate are shown initially. The shape of the `AudioIOTensor` is represented as `[samples, channels]`, which means the audio clip we loaded is mono channel with `28979` samples in `int16`."
```

Comment:
Interesting. it could be instructive to include an example that takes advantage of this lazyness. Does this work at all with `tf.data`?

Reply:
@MarkDaoust The lazy-loading works in graph mode and works in`tf.data` pipeline. This is one of the frequently requested features by user: Use `tf.data` as ETL and tensorflow-io will read data directly from a file format as part of the tf.data pipeline.

I am working on another tutorial to demo this type of usage. Will create a separate PR soon.